src/vm/runners/cairo_runner.zig

```diff
@@ -26,6 +26,76 @@ const expectEqual = std.testing.expectEqual;
 const expectError = std.testing.expectError;
 const expectEqualSlices = std.testing.expectEqualSlices;
 
+/// Tracks the step resources of a cairo execution run.
+const RunResources = struct {
+    // We consider the 'default' mode of RunResources having infinite steps.
+    n_steps: ?usize = null,
+
+    pub fn consumed(self: *RunResources) bool {
```

Comment:
Maybe we could define a `Self` type at the beginning of the struct to avoid repetitions of structure name, wdt?